src/main/java/bentobox/addon/islandfly/listeners/FlyListener.java

```diff
@@ -0,0 +1,69 @@
+package bentobox.addon.islandfly.listeners;
+
+import bentobox.addon.islandfly.FlySettings;
+import bentobox.addon.islandfly.IslandFlyAddon;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import world.bentobox.bentobox.BentoBox;
+import world.bentobox.bentobox.api.events.island.IslandEvent;
+import world.bentobox.bentobox.api.user.User;
+import world.bentobox.bentobox.managers.IslandsManager;
+
+import java.util.UUID;
+
+public class FlyListener implements Listener {
+
+    private final FlySettings settings;
+    private final BentoBox plugin;
+
+    public FlyListener(final IslandFlyAddon addon){
+        this.settings = addon.getSettings();
+        this.plugin = addon.getPlugin();
+    }
+
+    @EventHandler(ignoreCancelled = true)
+    public void onExitIsland(final IslandEvent.IslandExitEvent event){
+        // Check only when player exit is own island
+        final UUID playerUUID = event.getPlayerUUID();
+        if(!event.getIsland().getMembers().containsKey(playerUUID)) return;
+
+        // Player already flying
+        final User user = User.getInstance(playerUUID);
+        if(!user.getPlayer().getAllowFlight()) return;
+        // Bypass permission
+        if(user.hasPermission("islandfly.bypass")) return;
+
+        // Alert player fly will be disabled
+        final int flyTimeout = settings.getFlyTimeout();
+        user.sendMessage("islandfly.fly-outside-alert", "[timeout]", String.valueOf(flyTimeout));
```

Comment:
You should use the "text variable" `[number]`, like this:
```suggestion
        user.sendMessage("islandfly.fly-outside-alert", TextVariables.NUMBER, String.valueOf(flyTimeout));
```

It's not that important though, it's mainly about "consistency".
I let you choose for this one.